packages/botfuel-dialog/src/dialog-manager.js

```diff
@@ -14,23 +14,36 @@
  * limitations under the License.
  */
 
+// @flow
+
+import type { Config } from './config';
+import type { UserMessage, DialogData, DialogsData, MessageEntities } from './types';
+import type { BotMessageJson } from './messages/message';
+import type Bot from './bot';
+import type Brain from './brains/brain';
+import type ClassificationResult from './nlus/classification-result';
+import type { Action, ExecuteResult } from './dialogs/dialog';
+
 const logger = require('logtown')('DialogManager');
 const Resolver = require('./resolver');
 const Dialog = require('./dialogs/dialog');
 const DialogError = require('./errors/dialog-error');
 
+export type DialogManagerExecuteOutput = {|
```

Comment:
This is too complex...

Reply:
I agree. But without this type, a developper has to figure it out by him/herself each time she/he reads the code.

I think that this complex type is a strong smell that suggests refactoring.